lib/utils.coffee

```diff
@@ -306,6 +309,12 @@ exports.requestAsync = (options) ->
 			headers: options.headers
 			uri: urlLib.parse(url)
 		return response
+	.catch (error) ->
```

Comment:
I prefer the mutating style in that I read it more easily along the lines of "in this case I add an error handler, and in all cases I return this promise", rather than "in this case I add an error handler and return the promise, and in the other case I return the promise" - albeit they're equivalent in function and I would be ok with the ternary version.

As for the first one I suggested, in that style it's a bit ugly for sure so I'd go with one of the others, especially as it's not needed in this case at all because the predicate result is always known in advace.  However the way to make it a nicer would be to separate out the predicate function and have it named instead, ie:
```coffee
shouldRetry = -> retriesRemaining > 0
...
.catch shouldRetry, ->
	exports.requestAsync(options, retriesRemaining - 1)
```